File: src/com/gmail/marcosav2010/communicator/packet/packets/PacketFileAccept.java
package com.gmail.marcosav2010.communicator.packet.packets;

import java.io.IOException;

import com.gmail.marcosav2010.communicator.packet.Packet;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder;

public class PacketFileAccept extends Packet {
    
    private int fileId;
    
    public PacketFileAccept() {
	}
    
    public PacketFileAccept(int fileId) {
        this.fileId = fileId;
    }
    
    public int getFileId() {
    	return fileId;
    }
    
    @Override
    protected void encodeContent(PacketEncoder out) throws IOException {
    	out.write(fileId);
    }
    
    @Override
    protected void decodeContent(PacketDecoder in) throws IOException {
    	fileId = in.readInt();
    }
}

File: modules/file-transfer/src/main/java/com/gmail/marcosav2010/fth/packet/PacketFileAccept.kt
package com.gmail.marcosav2010.fth.packet

import com.gmail.marcosav2010.communicator.packet.Packet
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder
import kotlin.properties.Delegates.notNull

class PacketFileAccept() : Packet() {

    var fileID by notNull<Int>()
        private set

    constructor(fileID: Int) : this() {
        this.fileID = fileID
    }

    override fun encodeContent(out: PacketEncoder) {
        out.write(fileID)
    }

    override fun decodeContent(`in`: PacketDecoder) {
        fileID = `in`.readInt()
    }
}